### PR TITLE
add cn-cbor/1.0.0

### PR DESCRIPTION
**Pull request by gocarlos**
Specify library name and version:  **lib/1.0**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by gocarlos** [comment]
this is using the `complete` branch from @jimsch as as stated in https://github.com/cabo/cn-cbor/issues/60 the original author recommends to use it.

related to https://github.com/conan-io/conan-center-index/issues/1194

File: recipes/cn-cbor/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        extracted_dir = self.name + "-" + os.path.basename(self.conan_data["sources"][self.version]["url"]).split(".")[0]
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.definitions["fatal_warnings"] = False
+        self._cmake.definitions["coveralls"] = False
+        self._cmake.definitions["build_tests"] = False
```
Comment (uilianries) [inline recipes/cn-cbor/all/conanfile.py:57]:
```suggestion
        self._cmake.definitions["build_tests"] = False
        self._cmake.definitions["build_docs"] = False
        
```

File: recipes/cn-cbor/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        os.remove(os.path.join(self.package_folder, "LICENSE"))
+        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+        tools.rmdir(os.path.join(self.package_folder,
+                                 "lib", "cn-cbor", "cmake"))
+
+    def package_info(self):
+        self.cpp_info.libs = tools.collect_libs(self)
+        if self.settings.os == "Windows":
+            self.cpp_info.system_libs = ["ws2_32"]
```
Comment (uilianries) [inline recipes/cn-cbor/all/conanfile.py:81]: Why do you need ws2_32 ?

File: recipes/cn-cbor/all/conandata.yml [hunk 3]
```diff
@@ -0,0 +1,4 @@
+sources:
+  "20200326":
+    sha256: 7cc228a7ac56494f95b74dcf94c254f920aada736793cdf647f9a644f6c67c66
+    url: https://github.com/jimsch/cn-cbor/archive/e2b410487ab76a1b9ee639716c01622290a46dcb.zip
```
Comment (uilianries) [inline recipes/cn-cbor/all/conandata.yml:4]:
```suggestion
    url: https://github.com/jimsch/cn-cbor/archive/277e105f3ac52223408d35c25e01c0ddd0de217f.zip
```

Use the latest version available

File: recipes/cn-cbor/all/conandata.yml [hunk 4]
```diff
@@ -0,0 +1,4 @@
+sources:
+  "20200326":
```
Comment (uilianries) [inline recipes/cn-cbor/all/conandata.yml:2]:
```suggestion
  "20180730":
```

Use the date of commit

File: recipes/cn-cbor/all/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+from conans.errors import ConanInvalidConfiguration
+
+
+class CnCborStackConan(ConanFile):
+    name = "cn-cbor"
+    license = "MIT"
+    homepage = "https://github.com/jimsch/cn-cbor/"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = """A constrained node implementation of CBOR in C"""
+    topics = ("cbor")
```
Comment (uilianries) [inline recipes/cn-cbor/all/conanfile.py:12]:
```suggestion
    topics = ("cbor", "nodes", "messaging")
```

File: recipes/cn-cbor/all/conanfile.py [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.definitions["fatal_warnings"] = False
+        self._cmake.definitions["coveralls"] = False
+        self._cmake.definitions["build_tests"] = False
+        self._cmake.configure(build_folder=self._build_subfolder)
+        return self._cmake
+
+    def build(self):
+        cmake = self._configure_cmake()
```
Comment (uilianries) [inline recipes/cn-cbor/all/conanfile.py:62]: You need to patch the CMakeLists as it only builds shared library: https://github.com/jimsch/cn-cbor/blob/master/src/CMakeLists.txt#L19

File: recipes/cn-cbor/all/conanfile.py (hunk 2 above)
Comment (gocarlos) [reply recipes/cn-cbor/all/conanfile.py:81]: was in the cmake script

File: recipes/cn-cbor/all/conandata.yml (hunk 3 above)
Comment (gocarlos) [reply recipes/cn-cbor/all/conandata.yml:4]: no, the latest version available is not in the master branch, but in the complete branch...
the master branch is basically the same as https://github.com/cabo/cn-cbor

File: recipes/cn-cbor/all/conandata.yml (hunk 4 above)
Comment (gocarlos) [reply recipes/cn-cbor/all/conandata.yml:2]: see above

File: recipes/cn-cbor/all/conanfile.py (hunk 6 above)
Comment (gocarlos) [reply recipes/cn-cbor/all/conanfile.py:62]: https://github.com/jimsch/cn-cbor/blob/complete/CMakeLists.txt#L105

File: recipes/cn-cbor/all/conandata.yml (hunk 3 above)
Comment (uilianries) [reply recipes/cn-cbor/all/conandata.yml:4]: oh boy ... thanks for the information, now it explains all these things.

**Pull request merged**